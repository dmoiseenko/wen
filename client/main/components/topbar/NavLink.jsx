import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';


export default function NavigationLink({ route, label, openCloseMenu }) {
  function handleClickNavLink() {
    openCloseMenu();
  }

  return (
    <NavLink
      className="nav-item is-tab"
      activeClassName="is-active"
      exact
      to={route}
      onClick={handleClickNavLink}
    >
      {label}
    </NavLink>
  );
}

NavigationLink.propTypes = {
  route: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  openCloseMenu: PropTypes.func.isRequired
};

