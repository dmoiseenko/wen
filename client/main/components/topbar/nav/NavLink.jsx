import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';


export default function NavigationLink({ route, label, closeMenu }) {
  function handleClickNavLink() {
    closeMenu();
  }

  return (
    <NavLink
      className="nav-item is-tab"
      activeClassName="is-active"
      exact
      to={route}
      onClick={handleClickNavLink}
    >
      {label}
    </NavLink>
  );
}

NavigationLink.propTypes = {
  route: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  closeMenu: PropTypes.func.isRequired
};
