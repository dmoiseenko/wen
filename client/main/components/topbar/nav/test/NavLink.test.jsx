import React from 'react';
import { shallow } from 'enzyme';

import NavLink from '../NavLink';

const props = {
  route: 'route',
  label: 'label',
  closeMenu: jest.fn()
};

it('should render', () => {
  const wrapper = shallow(
    <NavLink {...props} />
  );

  expect(wrapper.node).toMatchSnapshot();
});

it('should call closeMenu when clicks on NavLink', () => {
  const closeMenu = jest.fn();

  const wrapper = shallow(
    <NavLink
      {...props}
      closeMenu={closeMenu}
    />
  );

  wrapper.find('.nav-item').simulate('click');

  expect(closeMenu).toHaveBeenCalled();
});
