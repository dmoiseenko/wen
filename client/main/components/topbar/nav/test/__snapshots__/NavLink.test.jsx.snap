// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<NavLink
  activeClassName="is-active"
  className="nav-item is-tab"
  exact={true}
  onClick={[Function]}
  to="route"
>
  label
</NavLink>
`;
